Arm/State.lean

```diff
@@ -662,3 +662,318 @@ theorem write_mem_irrelevant :
   simp [read_mem, write_mem, store_write_irrelevant]
 
 end Memory
+
+section NewMemory
+/-!
+# New definitions for the memory model
+
+For freedom in experimenting with definitions,
+we define our own version of `read_mem` and `write_mem`,
+called `read_mem'` and `write_mem`'.
+These operate directly on the memory, rather than the `ArmState`.
+We prove their equivalence to the existing definitions
+(`read_mem_eq_read_mem'`, `write_mem_eq_write_mem'`).
+
+Furthermore, we define equivalences to `BitVec.extractLsByte`
+to ease reasoning about byte-level manipulation.
+As an easy corollary, we get `getLsb` theorems on these to allow
+`omega` based reasoning about bit-level values of memory.
+-/
+
+abbrev Memory := Store (BitVec 64) (BitVec 8)
+
+/--
+A variant of `read_mem` that directly talks about writes to memory, instead of over the entire `ArmState`
+-/
+def read_mem' (addr : BitVec 64) (m : Memory) : BitVec 8 :=
+  read_store addr m
+
+theorem read_mem_eq_read_mem' : read_mem addr s = read_mem' addr s.mem := rfl
+
+@[simp]
```

Comment:
Why the `simp` tag? Time to have a memory-specific simp attribute? We typically do not use general-purpose `simp` lemmas in LNSym.

Reply:
Done.